docs/contributors/opentracingprovider.adoc

```diff
@@ -0,0 +1,203 @@
+To enable an end to end trace instance to be recorded and reconstructed at a later date, it is necessary to propagate trace state information between interacting services.
+
+=== Trace State
+
+The APM related trace state is comprised of:
+
+.Propagated state
+|===
+|Name |Description
+|traceId | The id associated with the trace instance being propagated
+|id | Unique id representing the interaction between the services
+|transaction | Optional name for the type of the trace instance
+|level | Optional field that governs what information should be reported for this trace instance (default is report all)
+|===
+
+The current reporting levels are:
+
+.Reporting levels
+|===
+|Level | Description
+|All | Report all trace information
+|None | Report no information (temporary decision)
+|Ignore | Long term decision that this trace type is not of interest
+|===
+
+=== Supported Transports
+
+Where a transport carries header properties with its messages, this can be used to propagate the trace state. The header property names are:
+
+.Propagated state
+|===
+|State | Header field name
+|traceId | HWKAPMTRACEID
+|id | HWKAPMID
+|transaction | HWKAPMTXN
+|level | HWKAPMLEVEL
+|===
+
+
+== Managing APM Trace Fragments
+
+=== Trace fragment completion
+
+As described in the following sections, Trace fragments are created under various situations, and a hierarchy of nodes (related to created spans) are constructed to represent the call trace for the fragment.
+
+
+Once all of the nodes have been completed, the Trace fragment can be submitted to the APM server. To detect when this situation occurs, it is recommended to use a form of reference counting, where the counter is incremented when a new Span is ‘started’ for the Trace fragment, and decremented when the Span is ‘finished’. So once the count reaches 0, the Trace fragment is in a state where it can be reported.
+
+
+=== Trace Recorder
+
+A Trace Recorder is responsible for dealing with completed Trace fragments.
+
+==== HTTP
+
+Each provider should support a HTTP based implementation of this recorder.
+
+
+
+QUESTION: [Batch size/interval - should each reporter support this, or should it be a common capability independent of the transport?]
+
+
+
+== Mapping Span to APM Node
+
+=== Processing References
+
+When requested to create a Span, the first step is to assess the supplied list of References (including ‘ChildOf’ references that may be provided separate from other e.g. ‘FollowsFrom’ references).
+
+==== Identify a Primary Reference
+
+A primary reference is the one that is considered the true ‘parent’ to the Span being created. It may be extracted SpanContext, a parent Span or a referenced to a FollowsFrom Span. Each will be handled in a different way - but first we will discuss how to identify whether a primary reference exists.
+
+
+First step is to identify the number of each type of reference, i.e.
+
+(a) References representing Extracted Trace State, whether ref type is ChildOf or FollowFrom
```

Comment:
`Extracted Trace State` why capitals, we may say that by extracted we mean `extractedContext = tracer.extract(carrier)`